File: Palindrome Number/PalindromeNumber.py
def isPalindrome(x: int) -> bool:
    if x<0:
        return False
    if x == 0:
        return True
    
    s = str(x)
    
    i = 0
    j = len(s)-1
    
    while i<j:
        if s[i] != s[j]:
            return False
        i += 1
        j -= 1
    
    return True

print(isPalindrome(121))

File: Palindrome Number/PalindromeNumber.php
<?php

function isPalindrome($x) {
    if ($x<0)
        return false;
    if ($x == 0)
        return true;
    $x = strval($x);
    $i = 0;
    $j = strlen($x)-1;

    while ($i < $j) {
        if ($x[$i] != $x[$j]) {
            return false;
        }
        $i += 1;
        $j -= 1;
    }
    return true;
        
}

print(isPalindrome(10));

?>